kustomize/commands/localize/localize.go

```diff
@@ -62,6 +68,36 @@ kustomize localize https://github.com/kubernetes-sigs/kustomize//api/krusty/test
 			if err != nil {
 				return errors.Wrap(err)
 			}
+
+			if !f.noVerify {
+				buildBuffer.Reset()
+				err := buildCmd.RunE(buildCmd, []string{args.target})
+				if err != nil {
+					return errors.Wrap(err)
+				}
+
+				originalBuild := buildBuffer.String()
+
+				buildDst := dst
+				if f.scope != "" && f.scope != args.target {
+					buildDst = filepath.Join(dst, filepath.Base(args.target))
+				}
+
+				buildBuffer.Reset()
+				err = buildCmd.RunE(buildCmd, []string{buildDst})
+				if err != nil {
+					return errors.Wrap(err)
+				}
+
+				localizedBuild := buildBuffer.String()
+
+				if localizedBuild == originalBuild {
+					log.Println("VERIFICATION SUCCESS: kustomize build for target and newDir are the same after localization.")
+				} else {
+					log.Println("VERFICATION FAILED: kustomize build for target and newDir are different after localization.")
+				}
```

Comment:
Ah I see, we would need to stop the execution if the `kustomize build` command fails with the diff output and then exit there so we don't see the success message from `kustomize localize`. I can add that. The part where I'm confused is

>  This check, however, is not useful for certain `target`s, including those that need flags to build.

does this mean targets that need the `--scope` flag to build?

Reply:
My interpretation, based on the accompanying examples, is that this is related to targets that would require flags to be passed to the `kustomize build` command, such as `--enable-exec`, `--enable-helm`, or `--load-restrictor`, to name a few, since there is no easy way we could proxy those `build` flags into `localize`.